views/index.erb

```diff
@@ -1,41 +1,99 @@
+        <h3 class="box-header">learning Experience</h3> 
+          <p class="profile-value"><%= @profile["learning_experience"] %></p>
+      </div>
+  </div>
+  <div class="row">
+      <div class="profile-section box col-md-6">
+        <h3 class="box-header">ambition</h3>
+          <p class="profile-value"><%= @profile["ambition"] %></p>
+      </div>
+      <div class="profile-section box col-md-6">
+        <h3 class="box-header">rationale</h3>
+          <p class="profile-value"><%= @profile["rationale"] %></p>
+      </div>
   </div>
+</div>
 
-  <div class="box">
-    <h3>Edit Profile</h3>
+<div class="other container-fluid">
+  <div class="row">
+    <div id="stats" class="col-md-6 box">
+      <h3>Stats</h3>
+      <span id="word-count"></span> words found.
+    </div>
 
-    <form action="/update" method="POST">
-      <fieldset>
-        <div class="form-group">
-          <label for="trainee[about]">About</label>
-          <input type="text" name="trainee[about]" value="<%= @profile["about"] %>"/>
-        </div>
+    <div id="problems" class="col-md-6 box">
+      <h3>Programming Problems</h3>
+      <a href="/primes">My Prime Number Solution</a>
+    </div>
 
-        <!-- TODO - Add more fields so that your profile can be edited -->
+    <div class="col-md-12">
+        <small class="edit-profile-button" data-toggle="modal" data-target="#edit-modal">Edit Profile</small>
+    </div>
+</div>
 
-        <input type="submit" value="Update Profile" class="btn btn-primary" />
-      </fieldset>
-    </form>
+<div id="edit-modal" class="modal">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-body">
+        <form action="/update" method="POST">
+          <fieldset>
+            <div class="form-group">
+              <label for="trainee[about]">About</label>
+              <input class="form-control" type="text" name="trainee[about]" value="<%= @profile["about"] %>"/>
+              <label for="trainee[github]">Github Username</label>
+              <input class="form-control" type="text" name="trainee[github]" value="<%= @profile["github"] %>"/>
+              <label for="trainee[codecademy]">Codecademy Username</label>
+              <input class="form-control" type="text" name="trainee[codecademy]" value="<%= @profile["codecademy"] %>"/>
+              <label for="trainee[homepage]">Web Address</label>
+              <div class="input-group">
+                <div class="input-group-addon">http://</div>
+                <input class="form-control" type="text" name="trainee[homepage]" value="<%= @profile["homepage"] %>"/>
+              </div>
+            </div>
+            <input type="submit" value="Update Profile" class="btn btn-primary" />
+          </fieldset>
+        </form>
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+      </div>
+    </div>
   </div>
+</div>
+
+</div>
```

Comment:
It's hard to tell at a glance if your HTML is well-formed, due to the indentation.